extensions/objects/templates/CoordTwoDimEditor.js

```diff
@@ -16,82 +16,67 @@ oppia.directive('coordTwoDimEditor', [
+          lat: $scope.value[0],
+          lng: $scope.value[1],
+          zoom: 0
         };
-
-        var updateMarker = function(lat, lng) {
-          var latLng = new google.maps.LatLng(lat, lng);
-
-          $timeout(function() {
-            if ($scope.mapMarker) {
-              $scope.mapMarker.setPosition(latLng);
-            } else {
-              $scope.mapMarker = new google.maps.Marker({
-                map: $scope.map,
-                position: latLng
-              });
+        $scope.mapMarkers = {
+          mainMarker: {
+            lat: $scope.value[0],
+            lng: $scope.value[1],
+            focus: true,
+            draggable: true,
+            icon: {
+              iconUrl: UrlInterpolationService.getExtensionResourceUrl(
+                '/interactions/InteractiveMap/static/marker-icon.png'),
+              // The size of the icon image in pixels.
+              iconSize: [25, 41],
+              // The coordinates of the "tip" of the icon.
+              iconAnchor: [12, 41],
+              shadowUrl: UrlInterpolationService.getExtensionResourceUrl(
+                '/interactions/InteractiveMap/static/marker-shadow.png'),
+              // The size of the shadow image in pixels.
+              shadowSize: [41, 41],
+              // The coordinates of the "tip" of the shadow.
+              shadowAnchor: [13, 41],
+              // The URL to a retina sized version of the icon image.
+              // Used for Retina screen devices.
+              iconRetinaUrl: UrlInterpolationService.getExtensionResourceUrl(
+                '/interactions/InteractiveMap/static/marker-icon-2x.png'),
+              shadowRetinaUrl: UrlInterpolationService.getExtensionResourceUrl(
+                '/interactions/InteractiveMap/static/marker-shadow.png')
             }
-          }, 10);
-        };
-
-        $scope.$watch('value', function(newValue, oldValue) {
-          // A new rule has just been created.
-          if ($scope.value === '') {
-            $scope.value = [0.0, 0.0];
           }
-
-          if (!angular.equals(newValue, oldValue)) {
-            updateMarker(newValue[0], newValue[1]);
+        };
+        $scope.mapEvents = {
+          map: {
+            enable: ['click'],
+            logic: 'emit'
+          },
+          markers: {
+            enable: ['dragend'],
+            logic: 'emit'
           }
-        });
-
-        // A new rule has just been created.
-        if ($scope.value === '') {
-          $scope.value = [0.0, 0.0];
-        }
+        };
 
-        // This is required in order to avoid the following bug:
-        //   http://stackoverflow.com/q/18769287
-        $timeout(function() {
-          updateMarker($scope.value[0], $scope.value[1]);
-          if ($scope.map) {
-            google.maps.event.trigger($scope.map, 'resize');
-          }
-        }, 100);
+        $scope.$on('leafletDirectiveMap.coordTwoDimEditor.click',
+          function(event, args) {
```

Comment:
Typically we use evt rather than event (here and below)

Reply:
Done!